cli/core/pkg/interfaces/config_client_interface.go

```diff
@@ -5,20 +5,38 @@
 package interfaces
 
 import (
+	configapi "github.com/vmware-tanzu/tanzu-framework/cli/runtime/apis/config/v1alpha1"
 	"github.com/vmware-tanzu/tanzu-framework/cli/runtime/config"
 )
 
 //go:generate counterfeiter -o ../fakes/config_client_fake.go . ConfigClient
-type ConfigClient interface {
+type ConfigClientWrapper interface {
 	GetEnvConfigurations() map[string]string
+	StoreClientConfig(clientConfig *configapi.ClientConfig) error
+	AcquireTanzuConfigLock()
+	ReleaseTanzuConfigLock()
 }
 
-type configClientImpl struct{}
+type configClientWrapperImpl struct{}
 
-func NewConfigClient() ConfigClient {
-	return &configClientImpl{}
+func NewConfigClient() ConfigClientWrapper {
+	return &configClientWrapperImpl{}
 }
 
-func (cc *configClientImpl) GetEnvConfigurations() map[string]string {
+func (cc *configClientWrapperImpl) GetEnvConfigurations() map[string]string {
 	return config.GetEnvConfigurations()
 }
+
+func (cc *configClientWrapperImpl) AcquireTanzuConfigLock() {
+	config.AcquireTanzuConfigLock()
+}
+
+func (cc *configClientWrapperImpl) ReleaseTanzuConfigLock() {
+	config.ReleaseTanzuConfigLock()
+}
+
+func (cc *configClientWrapperImpl) StoreClientConfig(clientConfig *configapi.ClientConfig) error {
+	// TODO: Add Read/Write locking mechanism before updating the configuration
+	// Currently we are only acquiring the lock while updating the configuration
```

Comment:
nit: We should remove this TODO comment.

Reply:
it was an existing comment. That's why I kept it.

Reply:
We need to keep this comment but not here, it needs to be kept in `cli/core/pkg/auth/csp/grpc.go` and not in the wrapper.